src/validators/required-if.js

```diff
@@ -13,7 +13,8 @@ const validate = (val, ruleObj, propertyName, inputObj) => {
     return required.validate(val);
   }
 
-  if (!isNil(inputObj[targetProperty]) &&
+  if (!isNil(inputObj) &&
```

Comment:
If the inputObj is nil, the calling of inputObj[targetProperty] will return error.
e.g. validate({ a:['requiredIf:b:foo'] }, undefined) => TypeError: Cannot read property 'b' of undefined

This line is added to prevent this kind of error

Reply:
@kevinraja1001 can u add the test cases to make sure that this is handled properly?

Reply:
sure @sendyhalim